src/hct_mis_api/apps/payment/api/views.py

```diff
@@ -163,3 +169,45 @@ def _get_action_permission(self, action_name: str) -> Optional[str]:
             PaymentPlan.Action.REVIEW.name: Permissions.PM_ACCEPTANCE_PROCESS_FINANCIAL_REVIEW.name,
         }
         return action_to_permissions_map.get(action_name)
+
+
+class PaymentPlanSupportingDocumentUploadView(APIView):
```

Comment:
Maybe you could just use `ViewSet` instead of `View` and have the `upload` and `delete` and `get` in one `ViewSet`? I see no reason to put them separately, since it is the same model, just different action